cyflash/bootload.py

```diff
@@ -164,15 +167,14 @@ def check_metadata(self, data, downgrade, newapp):
 			return
 
 		# TODO: Make this less horribly hacky
-		# Fetch from last row of last flash array
-		metadata_row = data.arrays[max(data.arrays.keys())][self.row_ranges[max(data.arrays.keys())][1]]
+		metadata_row = data.arrays[0][self.row_ranges[0][1]]
```

Comment:
This change causes it to fetch from the last row of the first flash array; I'm fairly sure this will break targets with multiple flash arrays.